src/utils/PromptCreator.ts

```diff
@@ -6,7 +6,22 @@ import {
 } from "@/types";
 
 export class PromptCreator {
-    constructor(private apiCatalog: TestingFrameworkAPICatalog) {}
+    constructor(private apiCatalog: TestingFrameworkAPICatalog) {
+    }
+
+    extendAPICategories(newCategories: TestingFrameworkAPICatalogCategory[] | TestingFrameworkAPICatalogCategory): void {
+        const categoriesArray = Array.isArray(newCategories) ? newCategories : [newCategories];
+        for (const category of categoriesArray) {
+            const existingCategory = this.apiCatalog.categories.find((existingCategory) => existingCategory.title === category.title);
+            if (existingCategory) {
+                for (const item of category.items) {
+                    existingCategory.items.push(item);
+                }
+            } else {
+                this.apiCatalog.categories.push(category);
+            }
+        }
+    }
```

Comment:
1. too many indentations (third-level blocks). let's try to flatten the code:


```ts
extendAPICategories(
  newCategories: TestingFrameworkAPICatalogCategory[] | TestingFrameworkAPICatalogCategory
): void {
  const categories = Array.isArray(newCategories) ? newCategories : [newCategories];

  categories.forEach((category) => {
    const existingCategory = this.apiCatalog.categories.find(
      (c) => c.title === category.title
    );

    if (existingCategory) {
      existingCategory.items.push(...category.items);
    } else {
      this.apiCatalog.categories.push(category);
    }
  });
}
```

Reply:
2. I think it's better if we won't allow passing a single category as an argument for parameter that is called `categories`